app/src/pages/clinic.tsx

```diff
@@ -0,0 +1,118 @@
+import clinics from '@public/data/clinics.json'
+import type { GetServerSideProps, NextPage } from 'next'
+import { useTranslation } from 'next-i18next'
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
+import { ChangeEvent, FormEvent, useState } from 'react'
+import Image from 'next/image'
+
+import ButtonLink from '@components/ButtonLink'
+
+const Clinic: NextPage = () => {
+  const { t } = useTranslation('common')
+  const [filteredClinics, setFilteredClinics] = useState<typeof clinics>([])
+  const [selectedClinic, setSelectedClinic] = useState<
+    typeof clinics[0] | undefined
+  >(undefined)
+  const [search, setSearch] = useState('')
+
+  const handleSearch = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    setSelectedClinic(undefined)
+    const filtered = clinics.filter((clinic) => clinic.zip === search)
+
+    setFilteredClinics(filtered)
+  }
+
+  const handleSelection = (e: ChangeEvent<HTMLInputElement>) => {
+    const { value }: { value: string } = e.target
+    const clinicIndex = filteredClinics?.findIndex(
+      (clinic) => clinic.clinic === value
+    )
+
+    setSelectedClinic(filteredClinics[clinicIndex])
+  }
+
+  const selected = (clinic: typeof clinics[0]) => {
+    return clinic.clinic === selectedClinic?.clinic
+  }
+
```

Comment:
This approach to handling the search means that if js is disabled, the search functionality will break. We can move forward with this for now, but if so, we should create a ticket to create a more robust option that takes advantage of the server side rendering.